src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnection.java

```diff
@@ -185,21 +150,13 @@ private String getApiToken(String apiTokenId, Item item) {
 
     protected GitLabConnection readResolve() {
         if (connectionTimeout == null || readTimeout == null) {
-            return new GitLabConnection(
-                    name, url, apiTokenId, new AutodetectGitLabClientBuilder(), ignoreCertificateErrors, 10, 10);
+            return new GitLabConnection(name, url, apiTokenId, ignoreCertificateErrors, 10, 10);
         }
-        if (clientBuilder == null) {
-            return new GitLabConnection(
-                    name,
-                    url,
-                    apiTokenId,
-                    new AutodetectGitLabClientBuilder(),
-                    ignoreCertificateErrors,
-                    connectionTimeout,
-                    readTimeout);
+        if (gitLabApi == null) {
+            return new GitLabConnection(name, url, apiTokenId, ignoreCertificateErrors, connectionTimeout, readTimeout);
         }
-        if (clientCache == null) {
-            clientCache = new HashMap<>();
+        if (gitLabApiCache == null) {
+            gitLabApiCache = new HashMap<>();
```

Comment:
As described above, we cannot break compatibility nor is it desirable to drop support for autodetection. This should be reverted.